src/PoolManager.sol

```diff
@@ -135,35 +129,33 @@ contract PoolManager is IPoolManager, Fees, NoDelegateCall, Claims {
 
     /// @inheritdoc IPoolManager
     function lock(address lockTarget, bytes calldata data) external payable override returns (bytes memory result) {
-        Lockers.push(lockTarget, msg.sender);
+        // Get the lock caller because thats an EOA and is not user-controlable
+        if (Locker.isLocked()) revert LockedBy(Locker.getLocker());
+
+        Locker.setLockerAndCaller(lockTarget, msg.sender);
```

Comment:
are these even neceessary anymore? feel like now they're kinda like `tx.origin` so maybe not that useful to expose?

Reply:
if we dont have it, how will our router know who to pull tokens from?
like our `router.lockAcquired` is called, they do some swap, and whose tokens do they use to settle? In my head they have to use `lockCaller`

Reply:
True - but what about a situation where like:
- I call router to do swaps
- router initiates lock
- I'm lockcaller
- ... 
- some hook or token or something calls router 
- I'm still lockcaller

Reply:
Maybe lockCaller / lockTarget isn't actually useful anymore?

Reply:
Yeah I think we still want it out? Idk if we decided 100%? Happy to pull it out in this PR if you 2 agree, otherwise we can make it a future issue to discuss